framework/grid/ActionColumn.php

```diff
@@ -213,20 +206,103 @@ protected function renderDataCellContent($model, $key, $index)
         return preg_replace_callback('/\\{([\w\-\/]+)\\}/', function ($matches) use ($model, $key, $index) {
             $name = $matches[1];
 
-            if (isset($this->visibleButtons[$name])) {
-                $isVisible = $this->visibleButtons[$name] instanceof \Closure
-                    ? call_user_func($this->visibleButtons[$name], $model, $key, $index)
-                    : $this->visibleButtons[$name];
+            return $this->renderButton($name, $model, $key, $index);
+        }, $this->template);
+    }
+
+    /**
+     * Renders button.
+     * @param string $name button name.
+     * @param \yii\db\ActiveRecordInterface|array $model the data model.
+     * @param mixed $key the key associated with the data model.
+     * @param int $index the current row index.
+     * @return string rendered HTML.
+     * @throws InvalidConfigException on invalid button format.
+     * @since 2.1.0
+     */
+    protected function renderButton($name, $model, $key, $index)
+    {
+        if (isset($this->visibleButtons[$name])) {
+            $isVisible = $this->visibleButtons[$name] instanceof \Closure
+                ? call_user_func($this->visibleButtons[$name], $model, $key, $index)
+                : $this->visibleButtons[$name];
+        } else {
+            $isVisible = true;
+        }
+
+        if (!$isVisible || !isset($this->buttons[$name])) {
```

Comment:
Isn't it better to error at the very start of the method in case there is non-existing button used in template?

Reply:
We have never done such a thing.
If you are sure code behavior should be changed in this way - I will add an exception.

Reply:
That is case for incorrect templates only as far as I can see. I think it is better to error in this case, otherwise typos won't be hard to spot and fix.

Reply:
Let's do it.